docs/deployAutomationForFirewallCostOptimization.md

```diff
@@ -0,0 +1,427 @@
+az login --tenant {tenant Id}
+
+# Set the subscription desired    
+az account set --subscription {subscription name}
+
+pwsh -command "Install-Module -Name Az.Automation; New-AzAutomationRunbook -Type 'PowerShell' -AutomationAccountName {automation account name} -Name {runbook name} -ResourceGroupName {automation resource group name};"
+
+```
+
+#### Create Automation Schedule and Link Runbook
+
+  ```bash
+    
+    New-AzAutomationSchedule -AutomationAccountName $Automation_Account_Name -Name $schedule_Name -StartTime $start_Time -ExpiryTime $end_Time -DayInterval 1 -ResourceGroupName $Resource_Group_Name_for_Automation
+
+    $params = @{}
+    $params.Add("resource_Group_Name_with_Firewall", "$Resource_Group_Name_with_Firewall")
+    $params.Add("resource_Group_Name_for_Automation", "$Resource_Group_Name_for_Automation") 
+    $params.Add("automation_Account_Name", "$Automation_Account_Name")  
+    $params.Add("tenant_Id", "$tenant_Id")
+    $params.Add("resource_Group_Name_with_Alerts", "$Resource_Group_Name_with_Alerts")
+    $params.Add("subscription_Name", "$subscription_Name")
+    $params.Add("vnet_Name", "$vnet_Name")
+    $params.Add("firewall_Name", "$firewall_Name")
+    $params.Add("pip_Name1", "$pip_Name1")
+    $params.Add("pip_Name2", "$pip_Name2")
+    $params.Add("pip_Name_Default", "$pip_Name_Default")
+    $params.Add("managed_Identity_Name", "$managed_Identity_Name")
+    $params.Add("environment", "$Environment")
+    $params.Add("action", "$action")
+    $params.Add("location", "$Location")
+
+    Register-AzAutomationScheduledRunbook -Parameters $params -ResourceGroupName $Resource_Group_Name_for_Automation -AutomationAccountName $Automation_Account_Name -RunbookName $powerShell_Runbook_Name -ScheduleName $schedule_Name
+
+
+  ```
+
+<img src="./assets/automation/endAutomationScheduleLinked1Portal.png" alt="Create a User-Assigned Managed Identity for the Automation" />
+
+<img src="./assets/automation/createAutmationAccountSchedule1Portal.png" alt="Create a User-Assigned Managed Identity for the Automation" />
+
+<img src="./assets/automation/editRunbookScheduleLinked1Portal.png" alt="Create a User-Assigned Managed Identity for the Automation" />
+
+<img src="./assets/automation/endAutomationScheduleLinked1Portal.png" alt="Create a User-Assigned Managed Identity for the Automation" />
+
+1. LInk the Runbook to the schedules
+
+### ADVANCED - Automated Scripts to Run
+
+BEFORE continuing please make sure all requirements have been met in the section labeled [prerequisites]("#-prerequisites").
+
+1. [Create Automation Infrastructure (BASH script)]("./scripts/automatoin/createAutomationForFirewallCostOptimization.sh")
+  
+  The [createAutomationForFirewallCostOptimization.sh]("./scripts/automation/createautimationForFirewallCostOptimization.sh") script "dot sources" the [firewallAutomationForCostOptimization.variables.sh]("./scripts/automation/cfirewallAutomationForCostOptimization.variables.sh").  
```

Comment:
Thoughts on making use of the saveenv.sh script and reusing some of variable defined in the main README? For example, there is already `ASB_ENV`, `ASB_SUBSCRIPTION_ID`, `ASB_TENANT_ID`, etc. The benefit there is that all the environment variables will end up in the same environment specific file. ex: `ngsa-asb-dev-eastus.env`

Reply:
I was struggling to have PowerShell read the envs that were set in BASH and vice versa, which is why I simplified.  I am glad to take another look though.